From f0b3e1be635c2d3e7cd6bd1400e8467c4877c652 Mon Sep 17 00:00:00 2001
From: S Mohan <s.mohan@xilinx.com>
Date: Fri, 6 Sep 2013 12:55:16 -0700
Subject: [PATCH 718/827] apf: Remove more sparse warnings

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Removed more sparse warnings based on input from Michal Simek

Signed-off-by: S Mohan <s.mohan@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit efaae7bd05012d52747a3c5f155b64cd7200b65e)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/staging/apf/xilinx-dma-apf.c | 11 +++++------
 drivers/staging/apf/xlnk.h           |  4 ++--
 2 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/apf/xilinx-dma-apf.c b/drivers/staging/apf/xilinx-dma-apf.c
index 99aae63..9ae372a 100644
--- a/drivers/staging/apf/xilinx-dma-apf.c
+++ b/drivers/staging/apf/xilinx-dma-apf.c
@@ -969,10 +969,10 @@ static int xdma_probe(struct platform_device *pdev)
 		return -EFAULT;
 	}
 
-	dev_info(&pdev->dev, "AXIDMA device %d physical base address=0x%08x\n",
-		 pdev->id, (unsigned int)res->start);
-	dev_info(&pdev->dev, "AXIDMA device %d remapped to 0x%08x\n",
-		 pdev->id, (unsigned int)xdev->regs);
+	dev_info(&pdev->dev, "AXIDMA device %d physical base address=%pa\n",
+		 pdev->id, &res->start);
+	dev_info(&pdev->dev, "AXIDMA device %d remapped to %pa\n",
+		 pdev->id, &xdev->regs);
 
 	/* Allocate the channels */
 
@@ -991,8 +991,7 @@ static int xdma_probe(struct platform_device *pdev)
 
 		/* Initialize channel parameters */
 		chan->id = i;
-		chan->regs = (void __iomem *)
-				((u32)xdev->regs + dma_chan_reg_offset);
+		chan->regs = xdev->regs + dma_chan_reg_offset;
 		/* chan->regs = xdev->regs; */
 		chan->dev = xdev->dev;
 		chan->max_len = XDMA_MAX_TRANS_LEN;
diff --git a/drivers/staging/apf/xlnk.h b/drivers/staging/apf/xlnk.h
index d3977bd..544d3bc 100644
--- a/drivers/staging/apf/xlnk.h
+++ b/drivers/staging/apf/xlnk.h
@@ -26,8 +26,8 @@ enum xlnk_dma_direction {
 union xlnk_args {
 	struct {
 		unsigned int len;
-		unsigned int *idptr;
-		unsigned int *phyaddrptr;
+		unsigned int __user *idptr;
+		unsigned int __user *phyaddrptr;
 		unsigned int cacheable;
 	} allocbuf;
 	struct {
-- 
2.9.3

